compiler-rt/lib/sanitizer_common/sanitizer_posix_libcdep.cpp

```diff
@@ -543,7 +543,11 @@ pid_t StartSubprocess(const char *program, const char *const argv[],
       internal_close(stderr_fd);
     }
 
+#  if SANITIZER_FREEBSD
+    internal_close_range(3, ~0U, 0);
```

Comment:
fd_t is a (signed) int, so ~0U is odd. Moreover, if you wanted to be generic across whatever integral type fd_t happens to be, ~0U will zero-extend to wider integers, regardless of the wider integer's type, and ~0 will sign-extend, regardless of the wider type. Or you can do `~(fd_t)0` if you want to be explicit and not have to think about integer conversion rules.

Reply:
+1, the types also look suspicious as is to me.

Reply:
I force-pushed since this is fairly trivial